aepp/configs.py

```diff
@@ -367,3 +370,141 @@ def setSandbox(self,sandbox:str=None)->dict:
+        if orgDevId is None and self.orgDevId is None:
+            raise ValueError("You are not providing the orgDevId and did not use the setOauthV2setup method.\n Use it to prepare this method")
+        if credentialId is None and self.credentialId is not None:
+            credentialId = self.credentialId
+        if orgDevId is None and self.orgDevId is not None:
+            orgDevId = self.orgDevId
+        if self.token is None:
+            raise Exception("You need to generate a token by using the connect method first")
+        myheader = {
+            'Authorization' : 'Bearer '+self.token,
+            'x-api-key' : self.client_id
+        }
+        endpoint = f"https://api.adobe.io/console/organizations/{orgDevId}/credentials/{credentialId}/secrets"
+        res = self.connector.getData(endpoint,headers=myheader)
+        return res
+    
+    def createSecret(self,credentialId:str=None,orgDevId:str=None)->dict:
+        """
+        Create a new secret with a new token for Oauth V2 credentials.
+        If you did not use the setOauthV2setup, you can pass the required information as parameters.
+        ATTENTION : In order to use it, you will need to have added the I/O Management API to your project.
+        Returns the new token and new secret that is automatically being used for that connection.
+        Arguments
+            credentialId : OPTIONAL : The credential id that can be found on your Project page.
+            orgDevId : OPTIONAL : the org Id but NOT the IMS string. It is defined on your project page.
+                Example : https://developer.adobe.com/console/projects/<orgId>/<projectId>/credentials/<credentialId>/details/oauthservertoserver
+        """
+        if self.connectionType != "oauthV2":
+            raise Exception("You are trying to use a service that is only supportede for OauthV2 authen. We do not support the other auth types.")
+        if credentialId is None and self.credentialId is None:
+            raise ValueError("You are not providing the credential ID and did not use the setOauthV2setup method.\n Use it to prepare this method")
+        if orgDevId is None and self.orgDevId is None:
+            raise ValueError("You are not providing the orgDevId and did not use the setOauthV2setup method.\n Use it to prepare this method")
+        if credentialId is None and self.credentialId is not None:
+            credentialId = self.credentialId
+        if orgDevId is None and self.orgDevId is not None:
+            orgDevId = self.orgDevId
+        if self.token is None:
+            raise Exception("You need to generate a token by using the connect method first")
+        myheader = {
+            'Authorization' : 'Bearer '+self.token,
+            'x-api-key' : self.client_id
+        }
+        endpoint = f"https://api.adobe.io/console/organizations/{orgDevId}/credentials/{credentialId}/secrets"
```

Comment:
will this endpoint be the same for stage and prod?

Reply:
Good question. I am only able to follow what is actually providing on the official documentation.  <https://developer.adobe.com/developer-console/docs/guides/authentication/ServerToServerAuthentication/implementation/>

I do not think that you are using OauthV2, but Oauth V1. 
I would suggest that we go live so provide clients with this functionality and if you ever need it and it is required to have a different url, we can use a variable to modify the endpoint automatically.